### DISCUSSION TITLE
Webshop: more filter options

### DISCUSSION THREAD

**Discussion by babsgosgens** [gameserverapp/Platform#674] [Suggestions]
Label filters are currently only available on parent level. Would be 'nice to have' if we could have these options also:

- Include labels from subpackages to filter options which would make subpackages easier to find,
- Include the filter in collection pages so collections can be used as a 'folder' to organise items.

![image](https://github.com/gameserverapp/Platform/assets/1268142/b121da2a-f481-467e-ad95-ab21dca649db)

**Comment by garmouchy** [comment]
I would chip in $20 for this option to be incorporated!

**Comment by babsgosgens** [comment]
Update:

On the shop.show route, the collection object that is passed to the view is similar to the object on the shop.index route.
Instead of items, it has children.

My suggestion would be to enable filtering (through the existing query parameters) on the returned child items. In addition it would be 'nice to have' if the route could call the paginatedResponse() function to enable pagination on them as well.

This example would only return child packs wich have "fiber" in their title. 
```/shop/shop-pack-id-collection/show?search=fiber&cluster=&gameserver=&filter=```
```php
return ShopTransformer::transform($this->api()->authRequest('get', 'shop/' . $id, [
            'children' => $query
        ], false));
```
_(You get the idea...)_

Is that feasible?

**Comment by mvaessen** [comment]
This is now available on your community website. The searching & filtering is done on the client-side. Meaning your browser is doing the filtering. 

Let me know if you experience any performance issues or other issues.

**Comment by garmouchy** [comment]
Hello,

So I said I would chip in $20… please let me know where to send it. If PayPal please send the email and if I can pay by visa send me the link and thank you so much for this awesome addition to the beautiful service you guys are offering. I know $20 is nothing for you guys but it’s a tiny percentage of a token of huge appreciation.

Thank you,
Hussam A. J.

On Oct 17, 2024, at 4:18 PM, Max Vaessen ***@***.***> wrote:

﻿

This is now available on your community website. The searching & filtering is done on the client-side. Meaning your browser is doing the filtering.

Let me know if you experience any performance issues or other issues.

**Reply by mvaessen** [reply]
Hi, thanks for the offer! I appreciate that. No need to send me money. If you want to support development then please consider buying an add-on on the dashboard.

If you already have an active add-on, than you are already supporting development. Thanks for that!

**Comment by garmouchy** [comment]
Yes I have been paying a steady 29€ per month :)

My pleasure!

Thank you,
Hussam A. J.

On Oct 17, 2024, at 4:48 PM, Max Vaessen ***@***.***> wrote:

﻿

Hi, thanks for the offer! I appreciate that. No need to send me money. If you want to support development then please consider buying an add-on on the dashboard.

If you already have an active add-on, than you are already supporting development. Thanks for that!

**Comment by babsgosgens** [comment]
Hi, Thank you so much for adding these Max. The filters from withing a collection work nicely.
 
 Re: Include labels from subpackages to filter options which would make subpackages easier to find,
 
 I am still missing the option to a) filter b) search for items that are in a collection from the parent level.
 
a) filter
As an example, when on parent-level `/shop`, when a collection "Resources" contains sub packages that are labelled "Scorched Earth", I would expect to see that label on the parent level. When selecting it, it would bring up the collection that contains those items.
 
b) search
As an example, when in a collection "Resources" that contains "metal ingots", I would expect to be able to drill down to the ingots when using the search term "metal"or "ingot" from within the collection.
Likewise, I would expect for those search terms o bring up the collection "Resources", as it contains metal ingots in a sub pack.

**Reply by mvaessen** [reply]
thanks for the additional info.

a: not sure whether i agree hat showing all child-labels on the main item is a good idea. When a collection has 30 items with all separate labels (i've seen them), then it becomes a mess.

b: can you provide screenshots, so i can better understand what you mean?

**Reply by babsgosgens** [reply]
I noticed that you have implemented a JS solution for searching within the collection. That is already quite helpful, ty. The one issue with this is that the search resets when you hit enter to activate the search. I think it would be better to open a new discussion in which I will outline some ideas.

Nevertheless, these are some mockups to illustrate the broader idea:

---
### Text search in collection (implemented with aforementioned JS solution):
![image](https://github.com/user-attachments/assets/663f6764-53f3-45c2-b07b-928dd31aad75)

---
### Text search on parent-level:
In the current implementation of the shop logic, this would bring up the collection(s) that contains any items that match the text search. In an ideal solution, the individual items would be displayed instead (will follow up on that at a later time in a new discussion):
![image](https://github.com/user-attachments/assets/236493a0-fda4-4762-b3e6-ee55a8af5122)


---
### Tag (filter) applied in current collection (implemented):
![image](https://github.com/user-attachments/assets/76437484-09ea-4bbb-85d4-0c74bc640c22)

---
### Tag (filter) applied on parent-level:
In the current implementation of the shop logic, this would bring up the collection(s) that contains any items that match the tag/filter. In an ideal solution, the individual items would be displayed instead (will follow up on that at a later time in a new discussion):
![image](https://github.com/user-attachments/assets/ddebd4be-977d-46a5-817d-8c84859527d5)

---

I hope that clarifies the general idea. If this is something that can be done in the current data setup, that would be great. If not, I understand.
